### PR TITLE
Task/jenkins 40956 icon button

### PR DESCRIPTION
**Pull request by cliffmeyers**
**Description**
Needed an easy way to place an icon inside of button, in support of the "SCM provider list" in the Creation UI
- New `IconButton` component
- New `monochrome` button style

**Submitter checklist**
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests

**Reviewer checklist**
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

**Comment by cliffmeyers** [comment]
Screenies:

Using material icons...
iconbutton-1-material-icons

Custom icons (svg)
iconbutton-2-custom-svg

Sizing
iconbutton-3-sizing

**Comment by cliffmeyers** [comment]
cc @sophistifunk as per usual :)

The one thing I'm not super happy with is the Material UI icons by nature have different shapes and it leads to some inconsistent padding on the left part of the button. As a workaround, I added two class names when material icons are used:
* `u-icon-${iconName}` - name of the specific material icon, in case someone wants to tweak the left padding on a button by button basis,
* `u-material-icon` - in case someone wants to apply a general fudge to all material icons

File: src/js/components/IconButton.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+ * @param {string} [className] - custom class name for outer element
+ * @param {object} [style] - custom style object
+ * @param {string} [label] - button text
+ * @param {string} [iconName] - name of material-ui icon to display
+ * @param {number} [iconSize] - width/height of icon
+ * @param {string} [iconFill] - color code to apply as fill
+ * @param {function} [onClick] - onclick callback function
+ * @constructor
+ */
+export function IconButton({children, className, style, label, iconName, iconSize, iconFill, onClick}) {
```
Comment (sophistifunk) [inline src/js/components/IconButton.jsx]: I'd really like to see some flow types for these sorts of config object params if nothing else.

**Comment by sophistifunk** [comment]
Besides me being a nagging type-grandma, es bueno 🐝

File: src/js/components/IconButton.jsx (hunk 1 above)
Comment (cliffmeyers) [inline src/js/components/IconButton.jsx]: Fixed in 1a6c96d

**Pull request merged**